app/controllers/HistoryController.scala

```diff
@@ -49,4 +49,20 @@ class HistoryController @Inject()(
             InternalServerError(s"Failed with the following error: $e")
         }
   }
+
+  def retrieveFirstDisclosure(arrangementId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      submissionDetailsRepository.retrieveFirstDisclosureForArrangementId(arrangementId).flatMap {
+        case Some(value) if value.disclosureID.isDefined =>
```

Comment:
if there is a reason the disclosure id might not be defined you'll need another case capturing it

Reply:
or turn the case None into a case _ if that's appropriate

Reply:
I originally had a case None but I removed it because I thought it would be unreachable. I've put it back just in case something happens to mongo, thanks.